### PR TITLE
Check Python typing for galaxy-containers and galaxy-tool-util during CI.

### PR DESCRIPTION
**Pull request by jmchilton**
With fixes - including a much cleaner new way to add methods to namedtuples I hadn't known about.

### CODE REVIEW

File: lib/galaxy/containers/__init__.py [hunk 1]
```diff
@@ -166,13 +167,13 @@ def map_port(self, port):
 
 class ContainerInterface(metaclass=ABCMeta):
 
-    container_type = None
-    container_class = None
-    volume_class = None
-    conf_defaults = {
+    container_type: Optional[str] = None
+    container_class: Optional[Type[Container]] = None
```
Comment (mvdbeek) [inline lib/galaxy/containers/__init__.py:171]: That's so helpful, I always struggle with what's expected here!

**Review by mvdbeek — APPROVED** [review]
Beautiful!

**Comment by nsoranzo** [comment]
> With fixes - including a much cleaner new way to add methods to namedtuples I hadn't known about.

That's a much nicer syntax, can we replace all `collections.namedtuple`s with `typing.NamedTuple` ?

**Comment by jmchilton** [comment]
> That's a much nicer syntax, can we replace all collections.namedtuples with typing.NamedTuple ?

Seems like a good plan, I'm not volunteering to do it but I'll hit the merge button if someone else does.

**Pull request merged**